### DISCUSSION TITLE
Prevent USB C Charging Hammering Battery

### DISCUSSION THREAD

**Discussion by ryanhossain9797** [seerge/g-helper#1030] [Ideas]
As far as I can see, USB C charging repeatedly goes on and off. The normal barrel jack as far as I can see stops charging after reaching full charge and waits till a good amount of battery drain before charging again.

Can this be implemented for USB C charging through GHelper? Maybe by hacking the charge limit Feature?

For example

say my normal battery charge limit is x%

When battery reaches x%, change limit to (x-5)%

When battery falls below (x-5)%, set limit back to x%

This way, if my original limit was say 100%, battery will stop charging at 100% and won't resume until it falls below 95%

**Comment by seerge** [comment]
Charging is controlled by hardware and BIOS. Even if you will adjust charging limits all the time, you would only make charge / discharge limits less frequent but longer :) so outcome will be same

Some people on reddit reported that official usb-c adapter allows bypassing battery on full charge.

**Reply by envyska** [reply]
Oh wait.. is offical usbc charger OK? Can I get some link for that? My stiff shoulder needs some false hope.

**Reply by ryanhossain9797** [reply]
I do too.

Does Asus make official USB C Adapters?

**Reply by IceStormNG** [reply]
They do. The ROG 100W USB-C charger. 
![image](https://github.com/seerge/g-helper/assets/12786283/88477e9a-ba76-4e66-9530-5bebb7890ffb)



Though: On my 2021 M16, it did not bypass the battery. On my 2023, it does. So this might only work properly for newer laptops. Or there was a firmware bug in my 2021 M16's EC (wouldn't surprise me).

**Reply by TheBestPessimist** [reply]
On my Flow X16 2023  USBC charging does not bypass battery. My battery always cycles charging <-> discharging, and in ~5 months the battery health is at 89%.

I'm pretty disappointed about the battery management.

**Reply by seerge** [reply]
@TheBestPessimist using asus charger?

**Reply by TheBestPessimist** [reply]
yep . the very one i received with the laptop.

**Reply by IceStormNG** [reply]
> On my Flow X16 2023 USBC charging does not bypass battery. My battery always cycles charging <-> discharging, and in ~5 months the battery health is at 89%.

This also happened on my 2021 M16. Interestingly my 2023 M16 doesn't do that anymore. It's still baffling me that they fixed it apparently on the M16 but not on the Flow. This makes no sense to me. But maybe it is also not fixed on the M16 and it is just a visual bug in the charge-controller/BIOS.

**Reply by riclun80** [reply]
My 2023 M16 does not bypass the battery on USB-C. Checked with a meter at the power outlet, 0-watt usage from the outlet with USB-C and with the Asus AC adapter it's 30+-watt.
Tested with 2 different GaN chargers, Ugreen 100w and Rocoren 140W.

**Reply by ryanhossain9797** [reply]
All I'm getting from this is nobody is sure what any of Asus' ROG Gaming laptops are doing in regards to USB C charging. Great job Asus, you're still getting this wrong in 2023. *Slow Clap*

Anyone have any firsthand info for 2023 Zephyrus G16?

EDIT: *Zephyrus G16 instead of G16, ambiguous with Strix G16*

**Reply by IceStormNG** [reply]
@riclun80 I just tested it again with my ASUS 100W ROG Charger and a USB-C power meter. The power draw is always around 12W (20V/0.8-0.9A) or more. When I run a benchmark, the amperage immediately jump to around 4-5 Amps, and falls back to 0.8-0.9 amps asap when I stop the benchmark. This looks to me that the power is passed through.

**Reply by IceStormNG** [reply]
I also tested with with an Anker 60W Charger. The Power always sits at 0.8A and 20V at least and jumps with the load. The power goes right to the components on mine. Works on both the left and the right port.

**Reply by IceStormNG** [reply]
> Anyone have any firsthand info for 2023 G16?

Do you mean the Zephyrus G16 or the Strix G16?

**Reply by ryanhossain9797** [reply]
> > Anyone have any firsthand info for 2023 G16?
> 
> Do you mean the Zephyrus G16 or the Strix G16?

Sorry my bad

Zephyrus G16 - GU603VV

**Reply by IceStormNG** [reply]
> Zephyrus G16 - GU603VV

The G16 is an 2022 M16 chassis with 2023 M16 components. Though I have no first hand experience with this one. My guess is, that it should work like on the M16, and it might be dependent on the charger and whether the cable has an E marker.
Of course, when the power draw goes above what the charger can supply, the battery will be drained.

**Reply by ryanhossain9797** [reply]
> > Zephyrus G16 - GU603VV
> 
> The G16 is an 2022 M16 chassis with 2023 M16 components. Though I have no first hand experience with this one. My guess is, that it should work like on the M16, and it might be dependent on the charger and whether the cable has an E marker.
> Of course, when the power draw goes above what the charger can supply, the battery will be drained.

Yeah I get your point. The M16 and G16 are basically the same things, The M16s are just the more higher end lineup of the G16

**Comment by riclun80** [comment]
@IceStormNG That's strange! If you charge it up to 60% and then place the charge limit to 50%, will it keep the charge level at 60% or go down to 50% after a while when you use usb-c power?

**Reply by ryanhossain9797** [reply]
I think it's going to go down to 50%
Which is not battery bypass, but it's better than it bouncing between 59% and 60%

**Reply by IceStormNG** [reply]
Let me test it... I will report back when I tested it. I first have to drain the battery a bit... cannot take too long with this power hungry machine.

**Reply by IceStormNG** [reply]
Alright. When I cap to 60%, it does some really weird stuff.

The spike was when I started a CPU stress test to put some load onto the machine. The flat area after that, is the duration of the stress test.
Otherwise it starts charging randomly, but the battery capacity barely changes... like more or less it stays the same.

<img width="773" alt="Screenshot 2023-08-11 153815" src="https://github.com/seerge/g-helper/assets/12786283/1b4d6070-939c-45f2-80a1-998207f206d6">

But: With the battery being full. it is constantly full with no change. This is idle on the first half and the second half is playing a game.
<img width="792" alt="Screenshot 2023-08-11 171138" src="https://github.com/seerge/g-helper/assets/12786283/5f414575-a911-46e3-a0ba-f44a62baf3fb">

No idea anymore how to interpret that.

**Reply by riclun80** [reply]
To me it looks like it's not bypassing the battery and behaving like mine.
What happened when you lowered the limit from 60 to 50 while still on USB-C power?

**Reply by IceStormNG** [reply]
<img width="437" alt="image" src="https://github.com/seerge/g-helper/assets/12786283/d056a002-e726-4c53-8b00-aebb9eb884f3">

I set it to 90% while on 100% charge. And yeah, it slowly discharges. Shit... and here I hoped ASUS would've finally got their crap together.

**Reply by ryanhossain9797** [reply]
I'm just glad framework is making a gaming laptop now. Asus is not the type to learn. They just make hyped up shiny products that look good on paper.

My next laptop is 100% going to be one of the framework 16inch laptops in 2 or 3 years.

**Reply by cosmicAP** [reply]
My type c charger (67 watt MI charger) stops charging about 10 minutes after plugging in , as far as i can assume the laptop don't communicate properly with charger and pulls full power from it in return the charger overheats and lowers the wattage can any one confirm if their laptop does the same (asus tuf dash f15 2022 76w/hr)
